Domains/3-DisciplineOther/Hydraulics/SewerHydraulicAnalysis.ecschema.xml

```diff
@@ -969,21 +996,88 @@
+        </Source>
+        <Target multiplicity="(0..*)" roleLabel="owned by" polymorphic="true">
+            <Class class="TimeOfConcentrationMethodAspect"/>
+        </Target>
+    </ECRelationshipClass>
+
+    <ECEntityClass typeName="TimeOfConcentrationMethodAspect" modifier="Abstract" displayLabel="TimeOfConcentrationMethodAspect">
+        <BaseClass>bis:ElementMultiAspect</BaseClass>
+        <ECProperty propertyName="HydraulicLength" typeName="double" displayLabel="Hydraulic Length" category="HydraulicData" kindOfQuantity="rru:LENGTH" description="The length of the flow travel path to use when calculating the Tc."/>
+        <ECProperty propertyName="Slope" typeName="double" displayLabel="Slope" category="HydraulicData" kindOfQuantity="rru:LENGTH" description="The average slope of the flow path to use when calculating the Tc."/>
+    </ECEntityClass>
+
+    <ECEntityClass typeName="ConveyanceTimeOfConcentrationMethodAspect" modifier="Abstract" displayLabel="ConveyanceTimeOfConcentrationMethodAspect">
+        <BaseClass>TimeOfConcentrationMethodAspect</BaseClass>
+        <ECProperty propertyName="ManningsN" typeName="double" displayLabel="Manning's n" category="HydraulicData" kindOfQuantity="rru:LENGTH" description="The Manning's n of the flow travel path to use when calculating the Tc."/>
+    </ECEntityClass>
+
+    <ECEntityClass typeName="TR55SheetFlowTcMethodAspect" modifier="Sealed" displayLabel="TR55SheetFlowTcMethodAspect">
+        <BaseClass>ConveyanceTimeOfConcentrationMethodAspect</BaseClass>
+        <ECProperty propertyName="TwoYearTwentyFourHourDepth" typeName="double" displayLabel="2 year 24 Hour Depth" category="HydraulicData" kindOfQuantity="rru:LENGTH_SHORT" description="The depth 2 year 24 hour storm at the location."/>
+    </ECEntityClass>
+
+    <ECEntityClass typeName="Tr55ShallowConcentratedFlowTcMethodAspect" modifier="Sealed" displayLabel="Tr55ShallowConcentratedFlowTcMethodAspect">
+        <BaseClass>TimeOfConcentrationMethodAspect</BaseClass>
+        <ECProperty propertyName="TreatAsPaved" typeName="boolean" displayLabel="Treat as Paved" category="HydraulicData" description="If true will use the paved form of the TR-55 Shallow Concentrated Flow Tc formula."/>
+    </ECEntityClass>
+
+    <ECEntityClass typeName="Tr55ChannelFlowTcMethodAspect" modifier="Sealed" displayLabel="Tr55ChannelFlowTcMethodAspect">
+        <BaseClass>ConveyanceTimeOfConcentrationMethodAspect</BaseClass>
+        <ECProperty propertyName="WettedPerimeter" typeName="double" displayLabel="Wetted Perimeter" category="HydraulicData" kindOfQuantity="rru:LENGTH" description="Wetted Perimeter of flow in channel to be used in TR-55 Channel Flow Formula for Tc."/>
+        <ECProperty propertyName="FlowArea" typeName="double" displayLabel="Flow Area" category="HydraulicData" kindOfQuantity="rru:LENGTH" description="Area of flow in channel to be used in TR-55 Channel Flow Formula for Tc."/>
```

Comment:
The property sounds to capture "Area" values, but its KoQ is "Length". Is that intended?